tests/test_telegram_topics.py

```
@@ -400,3 +400,77 @@ def test_use_topics_requires_state_store(topics_config):
+
+    with patch.object(bot._api, "create_forum_topic", side_effect=mock_create_topic):
+        with patch("auto_dev_loop.telegram.asyncio.sleep", side_effect=mock_sleep):
+            # Start first caller — it will hit RetryAfter and enter sleep
+            task1 = asyncio.create_task(bot._resolve_thread_id(sample_issue.repo))
+            await sleep_entered.wait()
+
+            # Second caller should NOT be blocked — it should be able to acquire the lock
+            # and succeed while the first caller is sleeping
+            task2 = asyncio.create_task(bot._resolve_thread_id(sample_issue.repo))
+            # Give task2 a chance to complete (it should succeed quickly)
+            done, pending = await asyncio.wait({task2}, timeout=0.5)
```

Ruff (0.15.5)

[warning] Unpacked variable `pending` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)